### PR TITLE
feat(mobile-ui): Collect frame delay metric

### PR DESCRIPTION
**Pull request by narsaynorath**
Adds a new metric for collecting the `frames.delay` property from a span's `data` object in the Starfish Mobile UI module. 

I've added the tags that I think are necessary to query and filter for this metric, and they are common filters for other metrics in this config. Also noted is that the data is in the `data` object, similar to #3473 and thus I modified measurement extraction to copy this value over to span measurements.

### CODE REVIEW

File: relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_mobile.snap [hunk 1]
```diff
@@ -74,7 +74,44 @@ expression: "(&event.value().unwrap().spans, metrics)"
...
+                ui_component_name: ~,
+                url_scheme: ~,
+                user: ~,
+                replay_id: ~,
+                sdk_name: ~,
+                other: {
+                    "frames.delay": I64(
+                        123,
+                    ),
+                },
```
Comment (narsaynorath) [inline relay-server/src/metrics_extraction/snapshots/relay_server__metrics_extraction__event__tests__extract_span_metrics_mobile.snap]: Am I doing something incorrect to have this field appear in the `other` property compared to the other keys?

**Review by iker-barriocanal — COMMENTED** [review]
Similar to https://github.com/getsentry/relay/pull/3473 -- is this doable through measurements?

File: relay-dynamic-config/src/defaults.rs [hunk 2]
```diff
@@ -622,6 +622,38 @@ fn span_metrics(
                     .always(), // already guarded by condition on metric
             ],
         },
+        MetricSpec {
+            category: DataCategory::Span,
+            mri: "d:spans/mobile.frames_delay@millisecond".into(),
```
Comment (markushi) [inline relay-dynamic-config/src/defaults.rs]: not sure if relevant, but as per develop spec, the SDKs send the frame delay [in seconds](https://develop.sentry.dev/sdk/performance/span-data-conventions/#mobile) right now

Comment (narsaynorath) [reply relay-dynamic-config/src/defaults.rs]: Thanks for catching that @markushi!

**Comment by narsaynorath** [comment]
Drafting until #3473 is settled, this PR will require changes similarly to that one.

File: relay-dynamic-config/src/defaults.rs [hunk 3]
```diff
@@ -727,6 +727,38 @@ fn span_metrics(
                     .always(),
             ],
         },
+        MetricSpec {
+            category: DataCategory::Span,
+            mri: "d:spans/mobile.frames_delay@second".into(),
```
Comment (jjbayer) [inline relay-dynamic-config/src/defaults.rs]:
```suggestion
            mri: "g:spans/mobile.frames_delay@second".into(),
```

**Review by jjbayer — COMMENTED** [review]
Should this be a gauge as well? Rest looks good to me!

File: relay-dynamic-config/src/defaults.rs (hunk 3 above)
Comment (iker-barriocanal) [inline relay-dynamic-config/src/defaults.rs]: Do gauges work, or do we need distributions?

Comment (narsaynorath) [reply relay-dynamic-config/src/defaults.rs]: Updated to use a gauge :+1:

Comment (narsaynorath) [reply relay-dynamic-config/src/defaults.rs]: Gauges work, I forgot to make that change here. Thanks!

**Review by jjbayer — APPROVED** [review]

**Pull request merged**